library/catalog/models.py

```diff
@@ -111,6 +114,28 @@ class Book(Document):
     store_links = ListField(default=[])
     statistic = EmbeddedDocumentField(BookStatistic.__name__, default=BookStatistic())
 
+    def calculate_rating(self):
```

Comment:
Yes, we need a better names for variables here.
And where did you get the rating calculation approach (formula) ?

Reply:
I wrote in doc function a short description and link to the source.

https://www.evanmiller.org/ranking-items-with-star-ratings.html

https://coderoad.ru/1411199/%D0%A7%D1%82%D0%BE-%D0%BC%D0%BE%D0%B6%D0%B5%D1%82-%D0%B1%D1%8B%D1%82%D1%8C-%D0%BB%D1%83%D1%87%D1%88%D0%B5-%D0%B4%D0%BB%D1%8F-%D1%81%D0%BE%D1%80%D1%82%D0%B8%D1%80%D0%BE%D0%B2%D0%BA%D0%B8-%D0%BF%D0%BE-5-%D0%B7%D0%B2%D0%B5%D0%B7%D0%B4%D0%BE%D1%87%D0%BD%D0%BE%D0%BC%D1%83-%D1%80%D0%B5%D0%B9%D1%82%D0%B8%D0%BD%D0%B3%D1%83